caproto/_hub.py

```diff
@@ -556,12 +556,7 @@ class _BaseChannel:
     def __init__(self, hub, name, address, priority, cid=None):
         self._hub = hub
         self.name = name
-        # Find an existing VirtualCircuit we can use or make a new one.
-        try:
-            circuit = self._hub.circuits[(address, priority)]
-        except KeyError:
-            circuit = VirtualCircuit(hub, address, priority)
-        self.circuit = circuit
+        self.circuit = self._hub.circuits[(address, priority)]
```

Comment:
In your pypvasync implementation (as last linked in GChat) you create the Circuit explicitly and stash in in `self._vc` and then use slicing (`[1:]`) to ignore the Channel's report of its own Circuit. I intended it to be possible for Clients to deal in Channels and ignore Circuits (in user code). Of course they still have to open up sockets, but this can be isolated to a separate part of the code, as roughly illustrated in the client tests. I can see an argument for this being too magical, but I want more time to think about it before we delete this.